app/main.py

```diff
@@ -1,3 +1,73 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float | int) -> None:
+        self.km = float(km)
+
+    def __str__(self) -> str:
+        km_str = int(self.km) if self.km.is_integer() else self.km
```

Comment:
This line is redundant, you could use `self.km` directly. Change the implementation of `__str__` and `__repr__` accordingly